### ISSUE TITLE
Get Segment Info + Update Branch Info (array dimension errors)

### ISSUE THREAD

**Issue by mjhyman** [BUNPC/psoct_vessel_graphing#2]
This [fork](https://github.com/mjhyman/psoct_vessel_graphing/tree/develop) contains a resolution to part 1 of this ticket. The same fork will contain the resolution to part 2 of this ticket.

**Part 1**: Get Segment Info
- There is a zero indexing issue in function nodeGrps_vesSegment.m. This occurs when running "verification > get segment info > update".

**Part 2**: Update Branch Info

The variable Data.Graph.segInfo.unverifiedIdx is initialized when calling Update Branch Info.  When it is initialized, it creates an array of zeros with a length equal to the number of segments (see below).

`if ~isfield(Data.Graph,'verifiedSegments')
        Data.Graph.verifiedSegments = zeros(size(Data.Graph.segInfo.segLen));
    end
    idx = find(Data.Graph.verifiedSegments == 0);
    Data.Graph.segInfo.unverifiedIdx = idx;`

After running "Update Branch Info," I then select a segment to delete and click the two subsequent pop-up windows to add it to the list of segments to delete. When I then click "Trash Deleted Segments," the code breaks in the function "verification_deleteSelectedSegments_Callback" on the line:
`Data.Graph.segInfo.unverifiedIdx = segMap(Data.Graph.segInfo.unverifiedIdx);`

The array "unverifiedIdx" has a length equal to the number of segments. However, the segMap has a smaller length. It appears that the array unverifiedIdx is not updated to reflect the true number of unverified indices. This bug is blocking the ability to delete segments.

**Comment by mjhyman** [comment]
To Do:

- Create separate branch from "mjhyman/psoct_vessel_graphing/[develop]" that only contains bugfix for zero index.
- Merge this zero index bugfix into main branch of origin repo.
- Create new branch for other bug in vesGraphValidate

**Comment by mjhyman** [comment]
Part 2 of this ticket was resolved in the latest push.

However, we have discovered a new bug. The variables "Data.Graph.segmentsGrpOrder" and "Data.Graph.segCGrps" are not updating properly after we "Trash deleted segments." These variables are not critical to the code running, but they are important for analyzing morphology. These variables are explained below.

_Data.Graph.segmentsGrpOrder_
This is an [N, 2] matrix. N is the total number of segments. The first column is the segment identification number and the second column is the group order (number of segments in group). A group is defined as one or more segments that are connected together. The group order is assigned in descending order of number of segments. For example, group 1 (out of 10) has the most segments whereas group 10 (out of 10) has the least amount of segments. A group can also contain just 1 segment.

_segCGrps_
This is [1, N] array where N is the total number of segments. The index of the array (i) corresponds to the segment ID and the value of the array [1, i] is the group ID. 

Typically, after running "Trash Deleted Segments," we then run "Update Branch Info" to automatically update the Graph parameters. However, "Update Branch Info" calls the line
`Data.Graph.segInfo.segmentsGrpOrder = segmentsGrpOrder;`
If X segments were deleted, then X last rows of this array will be set equal to 0.

Sreekanth will send me code for updating _segCGroups_ and _segmentsGrpOrder_. This code will be implemented in the section of code for "Trash Deleted Segments." The line above `Data.Graph.segInfo.segmentsGrpOrder = segmentsGrpOrder;` will be deleted.
